src/accelerate/test_utils/examples.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python
+
+# Copyright 2021 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import logging
+import os
+
+
+logger = logging.getLogger(__name__)
+
+
+def get_train_func(lines: list):
+    """
+    Finds the main training function from inside segmented source code.
+
+    Args:
+        lines (`list`):
+            Source code of a script
+    """
+    good_lines, found_start = [], False
+    for line in lines:
+        if not found_start and "def training_function" in line:
+            found_start = True
+            good_lines.append(line)
+            continue
+        if found_start:
+            if "def main" in line:
+                return good_lines
+            good_lines.append(line)
+
+
+def get_main_func(lines: list):
+    """
+    Finds the main function from inside segmented source code
+
+    Args:
+        lines (`list`):
+            Source code of a script
+    """
+    good_lines, found_start = [], False
+    for line in lines:
+        if not found_start and "def main" in line:
+            found_start = True
+            good_lines.append(line)
+            continue
+        if found_start:
+            if "if __name__" in line:
+                return good_lines
+            good_lines.append(line)
+
+
+def clean_lines(lines: list):
+    """
+    Filters `lines` and removes any entries that start with a comment ('#') or is just a newline ('\n')
+
+    Args:
+        lines (`list`):
+            Source code of a script
+    """
+    return [line for line in lines if not line.lstrip().startswith("#") and line != "\n"]
+
+
+def compare_against_test(base_filename: str, feature_filename: str, parser_only: bool, secondary_filename: str = None):
+    """
+    Checks whether the content aligned in `test_filename` is included inside of `full_filename`.
```

Comment:
Those do not match the signature. Also this could be expanded a little bit, I'm not sure I fully understand.